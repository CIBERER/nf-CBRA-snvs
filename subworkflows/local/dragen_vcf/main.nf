include { GATK4_CALIBRATEDRAGSTRMODEL                                       } from '../../../modules/nf-core/gatk4/calibratedragstrmodel/main'
include { GATK4_HAPLOTYPECALLER                                             }      from '../../../modules/nf-core/gatk4/haplotypecaller/main'

workflow DRAGEN_VCF {

    take:
    ch_bam         // channel (mandatory): [ val(meta), path(bam), path(bai) ]
    ch_fasta       // channel (mandatory): [ val(meta), path(fasta) ]
    ch_fai         // channel (mandatory): [ val(meta), path(fai) ]
    ch_refdict     // channel (mandatory): [ val(meta), path(dict) ]
    ch_ref_str     // channel (mandatory): [ val(meta), path(ref_str) ]
    ch_intervals    // channel (mandatory) : [ val(meta), path(bed) ]
    ch_dbsnp  // channel (mandatory) : [ val(meta3), path(vcf) ]
    ch_dbsnp_tbi  // channel (mandatory) : [ val(meta3), path(vcf) ]

    main:

    ch_versions = Channel.empty()
    
    GATK4_CALIBRATEDRAGSTRMODEL (
        ch_bam,
        ch_fasta.map { meta, fasta -> fasta },
        ch_fai.map { meta, fai -> fai },
        ch_refdict.map { meta, dict -> dict },
        ch_ref_str //.map { meta, ref_str -> ref_str }
    )
    ch_versions = ch_versions.mix(GATK4_CALIBRATEDRAGSTRMODEL.out.versions.first())

    ch_dragstr_model = GATK4_CALIBRATEDRAGSTRMODEL.out.dragstr_model
    ch_bam.view()
    ch_intervals.view()
    ch_dragstr_model.view()
    ch_bam.join(ch_intervals).join(ch_dragstr_model).view()

    GATK4_HAPLOTYPECALLER (
        ch_bam.join(ch_intervals).join(ch_dragstr_model),
        ch_fasta,
        ch_fai,
        ch_refdict,
        ch_dbsnp,
        ch_dbsnp_tbi
    )
    ch_versions = ch_versions.mix(GATK4_HAPLOTYPECALLER.out.versions.first())

    ch_dragstr_model = GATK4_HAPLOTYPECALLER.out.vcf

    emit:
    dragstr_model = ch_dragstr_model  // channel: [ val(meta), path(vcf) ]
    versions = ch_versions            // channel: [ versions.yml ]

}
