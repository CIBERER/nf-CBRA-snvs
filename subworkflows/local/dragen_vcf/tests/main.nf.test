nextflow_workflow {

    name "Test Subworkflow DRAGEN_VCF"
    script "../main.nf"
    workflow "DRAGEN_VCF"

    tag "subworkflows"
    tag "subworkflows/dragen_vcf"
    tag "gatk4"
    tag "gatk4/calibratedragstrmodel"

    setup {

        run("BWA_INDEX") {
            script "../../../../modules/nf-core/bwa/index/main.nf"
            process {
                """
                input[0] = [
                    [ id: 'genome'],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                ]
                """
            }
        }

        run("MAPPING") {
            script "../../../../subworkflows/local/mapping/main.nf"
            process {
                """
                input[0] = Channel.of([
                    [ id : 'test_sample' ],
                    [file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test.umi_1.fastq.gz', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test.umi_2.fastq.gz', checkIfExists: true)]
                ])
                input[1] =  Channel.fromList([
                            tuple([ id: 'test_sample'],
                            [])
                        ])
                input[2] = BWA_INDEX.out.index
                input[3] =  Channel.fromList([
                        tuple([ id: 'genome'],
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true))
                    ])
                input[4] =  Channel.fromList([
                        tuple([ id: 'genome'],
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta.fai', checkIfExists: true))
                    ])
                input[5] = Channel.fromList([
                        tuple([ id: 'genome'],
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.dict', checkIfExists: true))
                    ])
                input[6] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/vcf/dbsnp_146.hg38.vcf.gz', checkIfExists: true)
                input[7] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/vcf/dbsnp_146.hg38.vcf.gz.tbi', checkIfExists: true)
                """
            }
        }

        run("GATK4_COMPOSESTRTABLEFILE") {
            script "../../../../modules/nf-core/gatk4/composestrtablefile/main.nf"
            process {
                """
                input[0] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                input[1] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta.fai', checkIfExists: true)
                input[2] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.dict', checkIfExists: true)
                """
            }
        }
    }


    test("dragen_vcf - bam") {


        when {
            workflow {
                """
                input[0] = MAPPING.out.bam
                input[1] = Channel.fromList([
                        tuple([ id: 'genome'],file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true))
                ])
                input[2] =  Channel.fromList([
                    tuple([ id: 'genome'],file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta.fai', checkIfExists: true))
                    ])
                input[3] = Channel.fromList([
                    tuple([ id: 'genome'],file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.dict', checkIfExists: true))
                ])
                input[4] = GATK4_COMPOSESTRTABLEFILE.out.str_table.view()
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }

}
