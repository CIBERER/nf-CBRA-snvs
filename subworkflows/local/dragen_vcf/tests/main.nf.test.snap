{
    "dragen_vcf - bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample.biallelic.dragen.vcf.gz:md5,809e08e3cbd85bb73b4b6c09ecbfb2c6",
                        "test_sample.biallelic.dragen.vcf.gz.tbi:md5,688fa86277a1648520796452afb012c6"
                    ]
                ],
                "1": [
                    "versions.yml:md5,247abad81fc8bbdf3fae35db9746c49a",
                    "versions.yml:md5,620b7c3ab27a24487d87bb2e43e2248c",
                    "versions.yml:md5,7bbe20b5c60717428b83ce9fa9fe0274",
                    "versions.yml:md5,c62851f2cfad90ed92c37d82cd788431"
                ],
                "vcf": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample.biallelic.dragen.vcf.gz:md5,809e08e3cbd85bb73b4b6c09ecbfb2c6",
                        "test_sample.biallelic.dragen.vcf.gz.tbi:md5,688fa86277a1648520796452afb012c6"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,247abad81fc8bbdf3fae35db9746c49a",
                    "versions.yml:md5,620b7c3ab27a24487d87bb2e43e2248c",
                    "versions.yml:md5,7bbe20b5c60717428b83ce9fa9fe0274",
                    "versions.yml:md5,c62851f2cfad90ed92c37d82cd788431"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-05-05T08:31:07.888339101"
    }
}